osquery/tables/system/windows/prefetch.h

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#pragma once
+
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+struct PrefetchHeader {
+  int file_size;
+  std::string filename;
+  std::string prefetch_hash;
+};
+
+/**
+ * @brief Windows helper function to parse prefetch header data
+ *
+ * @returns Prefetch header data
+ */
+PrefetchHeader parseHeader(const std::string& prefetch_data);
```

Comment:
It seems like this function could fail to parse the input, so it should return an `Expected<Value, Error>` object instead

Reply:
there are several checks that are done before prefetch parsing starts (looking for prefetch signatures)
But this is a good idea, switched to `Expected`